cdc/redo/manager.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing pemissions and
+// limitations under the License.
+
+package redo
+
+import (
+	"context"
+	"math"
+	"sort"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/config"
+	cerror "github.com/pingcap/ticdc/pkg/errors"
+	"go.uber.org/zap"
+)
+
+var updateOffsetInterval = time.Second
+
+type consistentLevelType string
+
+const (
+	consistentLevelNomal    consistentLevelType = "normal"
+	consistentLevelEventual consistentLevelType = "eventual"
+)
+
+type consistentStorage string
+
+const (
+	consistentStorageLocal     consistentStorage = "local"
+	consistentStorageS3        consistentStorage = "s3"
```

Comment:
```suggestion
	// consistentStorageS3        consistentStorage = "s3"
```

Please do not add it until TiCDC actually supports it.

Reply:
It is a prototype PR, the first redo log writer we will add is s3, IMHO it is ok to leave it here